events/moose/moose.go

```diff
@@ -658,6 +687,222 @@ func (s *Subscriber) sendEvent(contentType, userAgent, requestBody string) int {
 	return errCodeEventSendSuccess
 }
 
+func (s *Subscriber) fetchSubscriptions() error {
+	if !s.enabled {
+		return nil
+	}
+	var cfg config.Config
+	if err := s.Config.Load(&cfg); err != nil {
+		return fmt.Errorf("loading config: %w", err)
+	}
+	token := cfg.TokensData[cfg.AutoConnectData.ID].Token
+
+	payments, err := s.SubscriptionAPI.Payments(token)
+	if err != nil {
+		return fmt.Errorf("fetching payments: %w", err)
+	}
+
+	orders, err := s.SubscriptionAPI.Orders(token)
+	if err != nil {
+		return fmt.Errorf("fetching orders: %w", err)
+	}
+
+	payment, ok := findPayment(payments)
+	if !ok {
+		return fmt.Errorf("no valid payments found for the user")
+	}
+
+	var orderErr error
+	order, ok := findOrder(payment, orders)
```

Comment:
I wanted to keep it consistent with `findPayment` and how go does it by default, e. g. with maps, where you do:
```go
	myMap := map[any]any{}
	val, ok := myMap[index]
```
`s.setSubscriptions` is never called with an empty order. Even if it is, it will not fill invalid data.